platforms/core-execution/workers/build.gradle.kts

```diff
@@ -20,7 +20,7 @@ dependencies {
     api(project(":serialization"))
     api(project(":service-provider"))
     api(project(":snapshots"))
-    api(project(":worker-processes"))
+    api(project(":worker-main"))
```

Comment:
It seems weird to me to see _anything_ (except maybe the distribution projects) depending on the `-main` projects. 

These `-main` projects are meant to represent applications. Why are we compiling against them? Shouldn't they be compiling against the rest of their dependencies so they are effectively the "root" of their respective dependency tree?

Reply:
Yes, absolutely. This PR doesn't try to detangle anything, just get the buckets in place. The "workers" project uses the "worker-main" project to find out the main class name so it can start the worker process. The same pattern is used for the other process too. I didn't try to change that in this PR.